src/rust/engine/src/context.rs

```diff
@@ -263,6 +263,7 @@ impl Core {
       let remote_execution_runner = Box::new(remote::CommandRunner::new(
         // We unwrap because global_options.py will have already validated this is defined.
         remoting_opts.execution_address.as_ref().unwrap(),
+        exec_strategy_opts.remote_cache_read,
```

Comment:
> #17195 already addresses that -- the issue is when remote execution servers themselves decide to read from the cache, agnostic to Pants's remote cache command runner. That is what this PR is focused on.

Right: but if there is a `remote_cache::CommandRunner` in use (which will always be the case when `--remote-cache-read=True`, then that code is what needs to implement the cache lookup: not the server (via `remote::CommandRunner`). If the server does it, then our logic will also be thwarted, because it will "look like" the server actually re-ran, when instead it just hit the cache.

Reply:
> but if there is a remote_cache::CommandRunner in use (which will always be the case when --remote-cache-read=True, then that code is what needs to implement the cache lookup: not the server (via remote::CommandRunner).

Which already happens. I'm not seeing the issue?

All that this PR does is make sure 100% that remote cache will not be used when `--no-remote-cache-read`. Note that the default fwict for `skip_cache_lookup` is False. Thus, zero changes in code when `--remote-cache-read`. I can double check that if you'd like to confirm though.